### DISCUSSION TITLE
Automatic configuration

### DISCUSSION THREAD

**Discussion by fredericoregateiro** [jstedfast/MailKit#1632] [Q&A] [answered]
First congratulations on the project, which fills a gap in .NET.

I know that what I'm going to talk about is outside the scope of this project, but I need some ideas and a push in the right direction.

Email configuration is not trivial, if I have the functionality to send emails in an application, as a rule, the common user does not know or understand how to obtain SMTP configuration settings.
Thunderbird lets you get the common settings and check if they are valid, just from the email address. How can I do something similar?
In the last step Thunderbird validates the password, how can I validate the password with the SMTP settings?

Thanks

**Comment by jstedfast** [comment, marked as answer]
Let me start with the easy question first :-)

> In the last step Thunderbird validates the password, how can I validate the password with the SMTP settings?

The only way to do this would be to actually login via the SmtpClient's Authenticate() method using the password - if it works, then the password is correct. If not, then it isn't.

> Thunderbird lets you get the common settings and check if they are valid, just from the email address. How can I do something similar?

I'm pretty sure that Thunderbird has a file hosted on their website somewhere that contains a list of domains and their IMAP, POP3 and SMTP configurations should default to. You might be able to just reuse their settings file.

I suspect that it's hosted on the web so that they can keep it updated and every version of Thunderbird that knows how to read that file can auto-update itself if/when that info changes.

The only reason I know this is because I just so happened to stumble onto it like 5+ years ago and thought it was kind of a cool idea. Then I forgot where that file was and haven't bothered to try and find it again.

Update:

Found info on the file format here: https://wiki.mozilla.org/Thunderbird:Autoconfiguration:ConfigFileFormat

More information can be found here (including info on how to use DNS queries to find some info): https://wiki.mozilla.org/Thunderbird:Autoconfiguration

It looks like you can find auto-config files by going to https://live.mozillamessaging.com/autoconfig/emailaddressdomain

... Aaaaaaand it looks like these URLs no longer work. It's just a default Apache installation now :-(

**Reply by fredericoregateiro** [reply]
Thanks for the reply and the all the help

**Reply by jstedfast** [reply]
Found a working solution:

It looks like you first want the following URL format: https://autoconfig.thunderbird.net/v1.1/{domain}

For example:

https://autoconfig.thunderbird.net/v1.1/gmail.com
